File: LinkedList-Stack-Queue/Q21.py
# https://leetcode.com/problems/merge-two-sorted-lists/

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        ptr = dummy = ListNode()
        while l1 and l2:
            if l1.val < l2.val:
                ptr.next = ListNode(l1.val)
                l1 = l1.next
            else:
                ptr.next = ListNode(l2.val)
                l2 = l2.next
            ptr = ptr.next
        ptr.next = l1 if l1 else l2
        return dummy.next

File: LinkedList-Stack-Queue/Q21.js
/**
 * https://leetcode.com/problems/merge-two-sorted-lists/
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
  const dummy = new ListNode();
  
  let ptr = dummy;
  while (list1 && list2) {
    if (list1.val < list2.val) {
      ptr.next = new ListNode(list1.val);
      list1 = list1.next;
    } else {
      ptr.next = new ListNode(list2.val);
      list2 = list2.next;
    }
    ptr = ptr.next;
  }
  ptr.next = list1 ? list1 : list2;
  
  return dummy.next;
};